app/scripts/views/recordtype/schema/edit-controller.js

```diff
@@ -105,8 +105,9 @@
                     disable_array_add: false,
                     theme: 'bootstrap3',
                     show_errors: 'change',
-                    no_additional_properties: true,
-                    startval: initialData
+                    startval: initialData,
+                    keep_oneof_values: false,
```

Comment:
`keep_oneof_values` helps prevent hidden metadata from persisting in the form when the user selects a different `type`.